controllers/dataprotection/backuprepo_controller.go

```diff
@@ -467,30 +570,46 @@ func (r *BackupRepoReconciler) createPVCForAssociatedBackups(
 }
 
 func (r *BackupRepoReconciler) checkOrCreatePVC(
-	reqCtx intctrlutil.RequestCtx, repo *dpv1alpha1.BackupRepo, namespace string) error {
+	reqCtx intctrlutil.RequestCtx, renderCtx renderContext,
+	repo *dpv1alpha1.BackupRepo, provider *storagev1alpha1.StorageProvider, namespace string) error {
+
 	pvc := &corev1.PersistentVolumeClaim{}
-	pvc.Name = repo.Status.BackupPVCName
-	pvc.Namespace = namespace
 	_, err := createObjectIfNotExist(reqCtx.Ctx, r.Client, pvc,
 		func() error {
-			storageClassName := repo.Status.GeneratedStorageClassName
-			volumeMode := corev1.PersistentVolumeFilesystem
-			resources := corev1.ResourceRequirements{}
-			if !repo.Spec.VolumeCapacity.IsZero() {
-				resources.Requests = corev1.ResourceList{
-					corev1.ResourceStorage: repo.Spec.VolumeCapacity,
+			if provider.Spec.PersistentVolumeClaimTemplate != "" {
+				// construct the PVC object by rendering the template
+				err := r.constructPVCByTemplate(renderCtx, pvc, repo, provider.Spec.PersistentVolumeClaimTemplate)
+				if err != nil {
+					return err
+				}
+			} else {
+				// set storage class name to PVC, other fields will be set with default value later
+				storageClassName := repo.Status.GeneratedStorageClassName
+				pvc.Spec = corev1.PersistentVolumeClaimSpec{
+					StorageClassName: &storageClassName,
 				}
 			}
-			pvc.Labels = map[string]string{
-				dataProtectionBackupRepoKey: repo.Name,
+			// overwrite PVC name and namespace
+			pvc.Name = repo.Status.BackupPVCName
+			pvc.Namespace = namespace
+			// add a referencing label
+			if pvc.Labels == nil {
+				pvc.Labels = make(map[string]string)
 			}
-			pvc.Spec = corev1.PersistentVolumeClaimSpec{
-				AccessModes: []corev1.PersistentVolumeAccessMode{
-					corev1.ReadWriteMany,
-				},
-				Resources:        resources,
-				StorageClassName: &storageClassName,
-				VolumeMode:       &volumeMode,
+			pvc.Labels[dataProtectionBackupRepoKey] = repo.Name
+			// set default values if not set
+			if len(pvc.Spec.AccessModes) == 0 {
+				pvc.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}
+			}
+			if pvc.Spec.VolumeMode == nil {
+				volumeMode := corev1.PersistentVolumeFilesystem
+				pvc.Spec.VolumeMode = &volumeMode
+			}
+			if pvc.Spec.Resources.Requests == nil {
+				pvc.Spec.Resources.Requests = corev1.ResourceList{}
+			}
+			if storageRequest := pvc.Spec.Resources.Requests.Storage(); storageRequest == nil || storageRequest.IsZero() {
```

Comment:
`storageRequest.IsZero() `contains nil check

Reply:
Fixed. It's because `Requests.Storage()` never returns nil.